### ISSUE TITLE
Status bar on/off trigger doesn´t work

### ISSUE THREAD

**Issue by Tialf** [kantlivelong/OctoPrint-PSUControl#106]
Hi,

it seems, that the statusbar trigger doesn´t work. If have connectet the signal cable to different GPIO-pins, but problem is still the same:

1. connecting signal to GPIO-pin
2. green led on relay is off -> no signal on the connected pin -> thats ok
3. changing in the configuration the GPIO-pin to the wired -> save -> trigger gets imediately a signal and turn the led into green, so this part of your software seems to work.
4. closing configuration
5. clicking on the statusbar icon for on/off -> it changes the symbol, but trigger is not working.

The GPIO-pin isn´t changing its status. If I manually plug in/out the cable it works, but not within your button :(
I´ve tested pin 3, 5, 7 and 11. Always the same game.

2018-07-01 21:58:36,848 - octoprint.environment - INFO - Detected environment is Python 2.7.13 under Linux (linux2). Details:
|  hardware:
|    cores: 4
|    freq: 1200.0
|    ram: 918192128
|  os:
|    id: linux
|    platform: linux2
|  plugins:
|    octopi_support:
|      model: 3B
|      revision: a02082
|      version: 0.15.1
|  python:
|    pip: 10.0.1
|    version: 2.7.13
|    virtualenv: /home/pi/oprint

PSU Control (0.1.7) = /home/pi/oprint/local/lib/python2.7/site-packages/octoprint_psucontrol
2018-07-02 20:31:00,761 - octoprint.plugins.psucontrol - INFO - Using GPIO for On/Off
2018-07-02 20:31:00,762 - octoprint.plugins.psucontrol - INFO - Using internal tracking for PSU on/off state.
2018-07-02 20:31:00,763 - octoprint.plugins.psucontrol - INFO - Running RPi.GPIO version 0.6.3
2018-07-02 20:31:00,764 - octoprint.plugins.psucontrol - INFO - Using GPIO for On/Off
2018-07-02 20:31:00,765 - octoprint.plugins.psucontrol - INFO - Configuring GPIO for pin 3

How can I get a valid log for you? Sry, I am not into such coding things ;)

I found something in the logging files:

2018-07-02 21:08:34,419 - octoprint.plugins.psucontrol - INFO - Switching PSU On
2018-07-02 21:08:37,604 - octoprint.plugins.psucontrol - INFO - Switching PSU Off
2018-07-02 21:08:45,880 - octoprint.plugins.psucontrol - INFO - Running RPi.GPIO version 0.6.3
2018-07-02 21:08:45,882 - octoprint.plugins.psucontrol - INFO - Using GPIO for On/Off
2018-07-02 21:08:45,883 - octoprint.plugins.psucontrol - INFO - Configuring GPIO for pin 7
2018-07-02 21:08:52,621 - octoprint.plugins.psucontrol - INFO - Switching PSU On
2018-07-02 21:08:57,067 - octoprint.plugins.psucontrol - INFO - Switching PSU Off
2018-07-02 21:08:59,048 - octoprint.plugins.psucontrol - INFO - Switching PSU On
2018-07-02 21:09:01,790 - octoprint.plugins.psucontrol - INFO - Switching PSU Off
2018-07-02 21:09:02,384 - octoprint.plugins.psucontrol - INFO - Switching PSU On
2018-07-02 21:09:03,861 - octoprint.plugins.psucontrol - INFO - Switching PSU Off
2018-07-02 21:09:04,798 - octoprint.plugins.psucontrol - INFO - Switching PSU On
2018-07-02 21:10:20,028 - octoprint.plugins.psucontrol - INFO - Switching PSU Off
2018-07-02 21:10:27,766 - octoprint.plugins.psucontrol - INFO - Running RPi.GPIO version 0.6.3
2018-07-02 21:10:27,768 - octoprint.plugins.psucontrol - INFO - Using GPIO for On/Off
2018-07-02 21:10:27,769 - octoprint.plugins.psucontrol - INFO - Configuring GPIO for pin 11
2018-07-02 21:12:26,728 - octoprint.plugins.psucontrol - INFO - Running RPi.GPIO version 0.6.3
2018-07-02 21:12:26,729 - octoprint.plugins.psucontrol - INFO - Using GPIO sensing to determine PSU on/off state.
2018-07-02 21:12:26,730 - octoprint.plugins.psucontrol - INFO - Configuring GPIO for pin 7
2018-07-02 21:12:26,731 - octoprint.plugins.psucontrol - INFO - Using GPIO for On/Off
2018-07-02 21:12:26,732 - octoprint.plugins.psucontrol - INFO - Configuring GPIO for pin 7
2018-07-02 21:12:32,658 - octoprint.plugins.psucontrol - INFO - Switching PSU Off
2018-07-02 21:12:33,580 - octoprint.plugins.psucontrol - INFO - Switching PSU Off
2018-07-02 21:12:33,759 - octoprint.plugins.psucontrol - INFO - Switching PSU Off
2018-07-02 21:12:34,192 - octoprint.plugins.psucontrol - INFO - Switching PSU Off
2018-07-02 21:12:34,363 - octoprint.plugins.psucontrol - INFO - Switching PSU Off
2018-07-02 21:12:34,556 - octoprint.plugins.psucontrol - INFO - Switching PSU Off
2018-07-02 21:12:35,236 - octoprint.plugins.psucontrol - INFO - Switching PSU Off
2018-07-02 21:12:51,366 - octoprint.plugins.psucontrol - INFO - Running RPi.GPIO version 0.6.3
2018-07-02 21:12:51,368 - octoprint.plugins.psucontrol - INFO - Using GPIO for On/Off
2018-07-02 21:12:51,369 - octoprint.plugins.psucontrol - INFO - Configuring GPIO for pin 7
2018-07-02 21:13:06,815 - octoprint.plugins.psucontrol - INFO - Switching PSU On
2018-07-02 21:13:08,671 - octoprint.plugins.psucontrol - INFO - Switching PSU Off
2018-07-02 21:13:09,546 - octoprint.plugins.psucontrol - INFO - Switching PSU On
2018-07-02 21:13:10,505 - octoprint.plugins.psucontrol - INFO - Switching PSU Off
2018-07-02 21:20:08,059 - octoprint.server.heartbeat - INFO - Server heartbeat <3
2018-07-02 21:35:03,275 - octoprint.plugins.psucontrol - INFO - Switching PSU On
2018-07-02 21:35:04,546 - octoprint.plugins.psucontrol - INFO - Switching PSU Off
2018-07-02 21:35:08,061 - octoprint.server.heartbeat - INFO - Server heartbeat <3
2018-07-02 21:35:40,321 - octoprint.plugins.psucontrol - INFO - Switching PSU On
2018-07-02 21:35:43,482 - octoprint.plugins.psucontrol - INFO - Switching PSU Off
2018-07-02 21:35:44,505 - octoprint.plugins.psucontrol - INFO - Switching PSU On
2018-07-02 21:35:46,082 - octoprint.plugins.psucontrol - INFO - Switching PSU Off
2018-07-02 21:44:47,086 - octoprint.plugins.psucontrol - INFO - Switching PSU On
2018-07-02 21:44:50,424 - octoprint.plugins.psucontrol - INFO - Switching PSU Off
2018-07-02 21:46:06,220 - octoprint.plugins.logging - INFO - Setting logger octoprint.plugins.psucontrol level to 10
2018-07-02 21:46:10,604 - octoprint.plugins.psucontrol - DEBUG - isPSUOn: False
2018-07-02 21:46:14,648 - octoprint.plugins.psucontrol - INFO - Switching PSU On
2018-07-02 21:46:14,650 - octoprint.plugins.psucontrol - DEBUG - Switching PSU On Using GPIO: 7
2018-07-02 21:46:14,757 - octoprint.plugins.psucontrol - DEBUG - isPSUOn: True
2018-07-02 21:46:17,013 - octoprint.plugins.psucontrol - INFO - Switching PSU Off
2018-07-02 21:46:17,014 - octoprint.plugins.psucontrol - DEBUG - Switching PSU Off Using GPIO: 7
2018-07-02 21:46:17,147 - octoprint.plugins.psucontrol - DEBUG - isPSUOn: False
2018-07-02 21:46:19,863 - octoprint.plugins.psucontrol - INFO - Switching PSU On
2018-07-02 21:46:19,864 - octoprint.plugins.psucontrol - DEBUG - Switching PSU On Using GPIO: 7
2018-07-02 21:46:19,973 - octoprint.plugins.psucontrol - DEBUG - isPSUOn: True
2018-07-02 21:46:22,515 - octoprint.plugins.psucontrol - INFO - Switching PSU Off
2018-07-02 21:46:22,516 - octoprint.plugins.psucontrol - DEBUG - Switching PSU Off Using GPIO: 7
2018-07-02 21:46:22,667 - octoprint.plugins.psucontrol - DEBUG - isPSUOn: False
2018-07-02 21:46:27,672 - octoprint.plugins.psucontrol - DEBUG - isPSUOn: False
2018-07-02 21:46:32,678 - octoprint.plugins.psucontrol - DEBUG - isPSUOn: False
2018-07-02 21:46:37,684 - octoprint.plugins.psucontrol - DEBUG - isPSUOn: False
2018-07-02 21:46:42,689 - octoprint.plugins.psucontrol - DEBUG - isPSUOn: False
2018-07-02 21:46:47,692 - octoprint.plugins.psucontrol - DEBUG - isPSUOn: False
2018-07-02 21:46:52,698 - octoprint.plugins.psucontrol - DEBUG - isPSUOn: False
2018-07-02 21:46:57,702 - octoprint.plugins.psucontrol - DEBUG - isPSUOn: False
2018-07-02 21:47:02,705 - octoprint.plugins.psucontrol - DEBUG - isPSUOn: False

Hope that helps.

**Comment by kantlivelong** [comment]
Hello,

You can get a fresh log by restarting OctoPrint then replicate the issue. Once done you can download the log from Settings->Logs

**Comment by daexys** [comment]
Hello,
I have similar problem. I am using jqc-3ff-s-z single relay. When I setup On/Off GPIO pin (40 in my case) printer tuns on and I cannot turn it off from status bar. To turn it off I must change my 40 pin to some other.

**Comment by vbz89** [comment]
having the same problem as described by OP

**Comment by kantlivelong** [comment]
@daexys @vbz89 Need logs.

**Comment by vbz89** [comment]
2018-07-28 20:07:56,706 - octoprint.plugins.psucontrol - INFO - Running RPi.GPIO version 0.6.3
2018-07-28 20:07:56,707 - octoprint.plugins.psucontrol - INFO - Using GPIO for On/Off
2018-07-28 20:07:56,708 - octoprint.plugins.psucontrol - INFO - Configuring GPIO for pin 7
2018-07-28 20:08:00,742 - octoprint.plugins.psucontrol - INFO - Switching PSU On
2018-07-28 20:08:04,247 - octoprint.plugins.psucontrol - INFO - Switching PSU Off
2018-07-28 20:08:13,564 - octoprint.plugins.psucontrol - INFO - Switching PSU On
2018-07-28 20:08:19,962 - octoprint.plugins.psucontrol - INFO - Switching PSU Off

Whenever I set the pin in the settings, the relay lights up. Toggling the power indicator does nothing. When I select a pin which I have not connected to the relay, the relay led goes off. 

Basically, the switching doesn't seem to be working with the indicator toggle.

I am also using a single relay module.

**Comment by kantlivelong** [comment]
It sounds like you have something wired incorrectly.

**Comment by vbz89** [comment]
Yup ok, I ran "gpio readall" in the terminal and the toggle indicator is working. There is something wrong with my wiring then. I'll let you know if I figure out what fixes it. Thanks

**Comment by kantlivelong** [comment]
@Spyder19 open a new ticket and provide the required info.

**Comment by kantlivelong** [comment]
Closing this ticket,
